### DISCUSSION TITLE
What is an example of a "list" ambiguous value in JIT? Are arbitrary box-shadows supported?

### DISCUSSION THREAD

**Discussion by littlee** [tailwindlabs/tailwindcss#5143] [Help]
in the doc
https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support

> The supported types are:
> - length
> - color
> - angle
> - list

How to use `list` exactly? What value can I pass as list?

BTW. Does jit mode support arbitrary box-shadow ?

```html
<div class="shadow-[0px 1px 2px 0px rgba(0, 0, 0, 0.04)]"></div>
```
the code above not work

**Comment by simonswiss** [comment]
Hey!

A `list` is a series of values, and is used for things like custom `grid-template-columns` (`grid-cols-[1fr,300px,1fr]`). These could be a `list` of `css` variables:

```html
<div class="grid-cols-[list:var(--col1),var(--col2),var(--col3)]">...</div>
```

---

As per why **you cannot do arbitrary box-shadows**:

On the very docs page you are linking, you can see that arbitrary values cannot contain spaces:

<img src="https://user-images.githubusercontent.com/485747/127968742-3eeef4a3-d837-416c-82b9-ce5b37c15319.png" width="700" />

---

You're supposed to replace spaces with commas instead.

The problem: the syntax for box-shadows contains _both_ spaces _and_ commas.... so the JIT engine wouldn't know how to break down that format without doing some consequent changes to how it interprets arbitrary values 😅

For custom shadows, best is to [implement those at the Tailwind config level, in the `boxShadow` key](https://tailwindcss.com/docs/box-shadow#customizing).

<img src="https://user-images.githubusercontent.com/485747/127969659-9d92b010-9493-4121-a14d-dc85da3e599b.png" width="700" />

Hope it helps!
